common/src/main/java/com/microsoft/identity/common/internal/net/ObjectMapper.java

```diff
@@ -133,7 +133,10 @@ public static String serializeObjectToFormUrlEncoded(Object object) throws Unsup
                     && !StringUtil.isEmpty((String)entry.getValue())) {
                 objectMap.put(URLDecoder.decode(entry.getKey(), ENCODING_SCHEME), URLDecoder.decode((String)entry.getValue(), ENCODING_SCHEME));
             } else {
-                objectMap.putAll(serializeNestedJsonToMap(entry.getValue().toString()));
+                final Map<String, String> hashMap = serializeNestedJsonToMap(entry.getValue().toString());
```

Comment:
Do these values need to be URL-decoded too?

Reply:
No, this will call serializeNestedJsonToMap iteratively where the values are URL-decoded in Line 134.